### DISCUSSION TITLE
Documentation for `uvicorn.{Server, Config}`

### DISCUSSION THREAD

**Discussion by Bibo-Joshi** [Kludex/uvicorn#1504] [Ideas] [answered]
Hi. I noticed that the classes `uvicorn.Server` and `uvicorn.Config` are not mentioned in the documentation over at uvicorn.org. In fact I only learned about them when I wanted to run uvicorn in the same process as other functionality and read the source of `uvicorn.run` to understand how that method works.

However, both classes seem to be intended for pulic usage, as they're both [declared in `__all__`](https://github.com/encode/uvicorn/blob/1a56da93744bd0b5a273960c0b7541759d5280e9/uvicorn/__init__.py#L5) and https://github.com/encode/uvicorn/pull/302 even mentiones the use cases.

IMO it would be very beneficial to have both classes documented along with the available parameters and accepted argument types :)

**Comment by Kludex** [comment, marked as answer]
This is now documented on https://www.uvicorn.org/#config-and-server-instances.
